source/common/network/io_socket_handle_impl.cc

```diff
@@ -103,6 +109,11 @@ Api::IoCallUint64Result IoSocketHandleImpl::read(Buffer::Instance& buffer, uint6
     bytes_to_commit -= slices[i].len_;
   }
   buffer.commit(slices, num_slices);
+
+  // Some tests try to read without initializing the file_event.
```

Comment:
Please protect all these "if (result.wouldBlock() && file_event_) {" with:
if constexpr ( PlatformDefaultTriggerType == FileTriggerType::EmulatedEdge) {

Reply:
see comment in https://github.com/envoyproxy/envoy/pull/13787#discussion_r526310514